scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -470,8 +470,10 @@ class Router(formatOps: FormatOps) {
           Split(NewlineT(isDouble = tok.hasBlankLine), 0)
         )
 
-      case tok @ FormatToken(_: T.RightParen, _: T.KwDef | _: T.Comment, _)
-          if leftOwner.is[Defn.ExtensionGroup] =>
+      case tok @ FormatToken(_: T.RightParen, _, _)
+          if leftOwner.is[Defn.ExtensionGroup] && nextNonComment(
+            formatToken
+          ).right.isNot[LeftParenOrBrace] =>
```

Comment:
should this be .is[KwDef]?

Reply:
So the change is actually related to `infix` as we can get `infix def` etc. So having it as `.isNot[LeftParenOrBrace]` is safer, because we want to match only if we don't have a `{` block or another `(` param list.

Reply:
is this cleaner than, say, .is[KwDef] or .is[Mod.xxx]?

Reply:
Yes, because we would need at least 3 with `Ident` as `infix` is a soft keyword. Also, it's more logical `LeftParenOrBrace` means we are not starting a block and not doing another param list, so the only other alternative is that it is the definition.

Reply:
thank you, understood. unless there's a more logical way with rightOwner tree, vs right token, I'm ok with this. thanks again for looking into this.